mempool/reactor.go

```diff
@@ -272,6 +272,8 @@ func (memR *Reactor) broadcastTxRoutine(peer p2p.Peer) {
 
 		// Do not send this transaction if we receive it from peer.
 		if entry.IsSender(peer.ID()) {
+			memR.Logger.Debug("Skipping transaction, peer is sender",
```

Comment:
Do we need all these debug lines?

Reply:
Keep in mind that every such line introduces an additional performance penalty.

Reply:
This is a good point, but currently we don't have any debug lines on this method, which is pretty important.

Reply:
What we can do here is to cache the costly part of this call, the `entry.Tx()` and re-use the value in all debug entries.

What do you think?

Reply:
Use `log.NewLazySprintf`? Good idea 👍

Reply:
Solved in https://github.com/cometbft/cometbft/pull/4127/commits/d3961f9d035e4d2c75535daf368d7e421fcf1260.

Reply:
However, we are still paying the cost of `tx.Hash()` even without debug.

This is the same as in the rest of this code. Namely, we aren't saving anything here.